src/Inciter/AMR/marked_refinements_store.h

```diff
@@ -0,0 +1,113 @@
+#ifndef AMR_marked_refinements_store_h
+#define AMR_marked_refinements_store_h
+
+#include <map>
+#include "Refinement_State.h"
+
+namespace AMR {
+
+    class marked_refinements_store_t {
+        private:
+            std::map<size_t, Refinement_Case> marked_refinements;
```

Comment:
In a second round, we should probably test if we can get away with `std::unordered_map`.